### PR TITLE
FIX: FPC 3.3 compatibility

### PR DESCRIPTION
**Pull request by larinsv**
The source file has been modified to allow doublecmd to be compiled with the current (trunk / main) version of FPC.  
At the same time, the project can also be compiled using FPC 3.2

### CODE REVIEW

File: plugins/wcx/unrar/src/unrarfunc.pas [hunk 1]
```diff
@@ -25,6 +25,7 @@
 unit UnRARFunc;
 
 {$mode objfpc}{$H+}
+{$modeswitch arraytodynarray}
```
Comment (larinsv) [inline plugins/wcx/unrar/src/unrarfunc.pas:28]: When compiling with FPC 3.2.2, a Warning occurs here:

> Warning: Illegal compiler switch "ARRAYTODYNARRAY+"

You can ignore this, or I can write something like this:

```pascal
{$ifndef VER3_2}
{$modeswitch arraytodynarray}
{$endif}
```

Comment (alexx2000) [reply plugins/wcx/unrar/src/unrarfunc.pas:28]: Yes, it is better to add IFDEF.

Comment (larinsv) [reply plugins/wcx/unrar/src/unrarfunc.pas:29]: Added IFDEF

**Comment by vvch** [comment]
Unfortunately compatibility with fpc 3.0.4 was lost with this commit which is not good since this version of fpc is the only available one from the official repository in some still actual linux distributions, e. g. Ubuntu 20.04

**Pull request merged**